core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -76,6 +89,12 @@ private static boolean isAggregation(FunctionCall functionCall, Session session,
                 || functionCall.getOrderBy().isPresent();
     }
 
+    private static boolean isWindow(FunctionCall functionCall, Session session, Metadata metadata)
```

Comment:
Seems like this method is going to be ambiguous with `isWindowFunction(FunctionCall)` and when it's appropriate to use one vs the other. The difference seems to be the delegated call down to `metadata.isWindowFunction(session, functionCall.getName())`, do we know which cases the current `functionCall.getWindow().isPresent()` behavior fails to identify correctly?

Reply:
`functionCall.getWindow().isPresent()` fails when the window function is used without `OVER()`. It seems that we update the window variable after checking for the existence of `OVER()`. This check is being done after the `verifyNoAggregateWindowOrGroupingFunctions()` call, hence the compiler failed issue.